capstone/src/main/java/com/google/sps/data/Tag.java

```diff
@@ -101,4 +101,14 @@ private Tag(String tag) {
     public String getTag() {
         return this.tag;
     }
+
+    public static final HashMap<String, Integer> constructUserVectorMap() {
+        HashMap<String, Integer> vector = new HashMap<>();
+
+        for(String tag : CATEGORY_TAGS) {
+            vector.put(tag, 0);
+        }
+        
+        return vector;
+    }
```

Comment:
(optional) I would recommend not having this. Instead, just create a new hashmap, and wherever you are using this, do checks to see if a key is already present instead of just getting it and knowing its there. Bugs could crop up if for example this only pre-fills categories but later down the road the reccomender takes into account length tags for example. Also, its unnecessary work to build up this map if for example only 2 categories are interacted with